simulator/src/eventQueue.js

```
@@ -16,7 +16,7 @@ export default class EventQueue {
     */
     add(obj, delay) {
         if (obj.queueProperties.inQueue) {
-            obj.queueProperties.time = this.time + (delay || obj.propagationDelay);
+            obj.queueProperties.time = this.time + (delay !== undefined ? delay : obj.propagationDelay);
```

ESLint

[error] Assignment to property of function parameter 'obj'.

(no-param-reassign)